### PR TITLE
Allow specifying BigQuery credential directly [sc-9738]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

Provides the option to specify the BigQuery credentials from the key file directly in the config, and bypass the key file upload.

### 🤓 What?

- Update BigQuery configs to take either key_path or credentials
- Update readme
- bump up bigquery library versions

### 🧪 Tested?

tested bigquery and bigquery usage crawlers locally using both config formats

### CODE REVIEW

File: metaphor/bigquery/utils.py [hunk 1]
```diff
@@ -25,29 +25,38 @@
...
-        credentials = service_account.Credentials.from_service_account_info(
-            json.load(fin),
-            scopes=["https://www.googleapis.com/auth/cloud-platform"],
-        )
-
-        return logging_v2.Client(
-            credentials=credentials,
-            project=project_id if project_id else credentials.project_id,
-        )
+    raise ValueError(
```
Comment (mars-lan) [inline metaphor/bigquery/utils.py:57]: Might be better to use a pydantic validator? https://pydantic-docs.helpmanual.io/usage/validators/

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM

File: metaphor/bigquery/utils.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/bigquery/utils.py]: good point, updated

Comment (mars-lan) [reply metaphor/bigquery/utils.py]: Root validator might be more appropriate here? https://pydantic-docs.helpmanual.io/usage/validators/#root-validators

Comment (alyiwang) [reply metaphor/bigquery/utils.py]: thanks, updated

**Pull request merged**